DATASETS_v2.md

```diff
@@ -0,0 +1,244 @@
+
+Those are all of the required fields. Certain dataTypes can have extra fields that help specify the type of data that the column should hold - see the "Extra type info" section below.
+
+##### Data types
+
+The following data types are supported by sno. When a versioned sno dataset is converted to a database table (ie, when `sno checkout` updates the working copy) then these sno data types will be converted to equivalent data types in the database table, depending on what is supported by the database in question.
+
+* boolean
+* blob
+* date
+* datetime
+* float
+* geometry
+* integer
+* interval
+* numeric
+* text
+* time
+* timestamp
+
+##### Extra type info
+
+Certain types have extra fields that help specify how the type should be stored in a database. They don't necessarily affect how sno stores the data - and they don't necessarily affect all database types - for instance, setting a maximum length of 10 characters in a column with `"dataType": "text"` won't be enforced in a SQLite since it doesn't enforce maximum lengths, and if a user then modifies a field in this column to contain an overlength string, this can still be committed to sno.
+
+The extra fields that are supported are as follows:
+
+###### Extra type info for `"dataType": "geometry"`
+- `geometryType`
+  * Eg `"geometryType": "MULTIPOLYGON ZM"`
+  * A WKT geometry type - eg "POINT", "LINESTRING", "MULTIPOLYGON", etc, optionally followed by a Z or M indicator if the data has a third dimension or a linear referencing system (or both).
+- `geometryCRS`
+  * Eg `"geometryCRS": "EPSG:2193"`
+  * A string used to identify the Coordinate Reference System of the geometry. Often in the form `"EPSG:1234"` for a CRS in the EPSG registry, but for a custom CRS, any identifier could be chosen.
+  * Can be `null` for an unspecified CRS.
+
+###### Extra type info for `"dataType": "integer"`
+- `size`
+  * Eg `"size": 16`
+  * The size of the integer in bits. Should be 16, 32, or 64.
+
+###### Extra type info for `"dataType": "float"`
+- `size`
+  * Eg `"size": 32`
+  * The size of the floating point number in bits. Should be 32 or 64.
+
+###### Extra type info for `"dataType": "text"`
+- `length`
+  * Eg: `"length": 100`
+  * The maximum length of the text in characters.
+  * Can be null if the maximum length is unbounded.
+
+###### Extra type info for `"dataType": "numeric"`
+- `precision`
+  * The maximum number of digits for the numeric type. For 
+- `scale`
+  * How many of the digits are to the right of the decimal point.
+For example, the number "1234.5678" can be stored in a numeric type with a precision of 8 and a scale of 4.
+
+###### Extra type info for `"dataType": "timestamp"`
+- `timeZone`
```

Comment:
Not really sure: perhaps there's a bug in your schema.json gist, which has timezone as metadata you can add to an entire column:
https://gist.github.com/rcoup/2774c12af9f3997088dd13a8a84eff98#file-schema-js

Perhaps that's correct, or perhaps the metadata we need is not the timezone, but whether the column's complete type is `TIMESTAMP WITH TIMEZONE` or `TIMESTAMP WITHOUT TIMEZONE`.

GPKG doesn't have timestamps and no extra work has been done on them so this spec is actually only aspirational at this point. I've just added bug tracking this: https://github.com/koordinates/sno/issues/225
So, I've just taken the extra type info section out for timestamp. I've left timestamp in the list of types, since we will support it, but it's actually not supported as of 0.5